spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindFieldSelfAssignment.java

```diff
@@ -161,7 +167,28 @@ public void sawOpcode(int seen) {
         default:
             break;
         }
+        if (seen == Const.GETFIELD) {
+            XField f = getXFieldOperand();
+            OpcodeStack.Item top = stack.getStackItem(0);
+            XField topF = top.getXField();
+            if (f == null || topF == null) {
+                return;
+            }
 
+            if ("this$0".equals(topF.getName())) {
+                parentInstanceLoadFromRegister = top.getFieldLoadedFromRegister();
+            }
+        }
     }
 
+    @Override
+    public void afterOpcode(int seen) {
+        super.afterOpcode(seen);
+
+        if (seen == Const.GETFIELD && parentInstanceLoadFromRegister > -1) {
+            OpcodeStack.Item top = stack.getStackItem(0);
+            top.setUserValue(parentInstanceLoadFromRegister);
```

Comment:
when you use `UserValue` in your detector, please annotate the detector with `@CustomUserValue`.

refs https://blog.kengo-toda.jp/entry/2013/10/13/011552